unittest/client/src/ilm_common_unittests.cpp

```diff
@@ -0,0 +1,374 @@
+/***************************************************************************
+ *
+ * Copyright (C) 2023 Advanced Driver Information Technology Joint Venture GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <gtest/gtest.h>
+#include "ilm_common.h"
+#include "ilm_common_platform.h"
+#include "client_api_fake.h"
+
+extern "C"
+{
+    FAKE_VALUE_FUNC(ilmErrorTypes, ilmControl_init, t_ilm_nativedisplay);
+    FAKE_VOID_FUNC(ilmControl_destroy);
+    FAKE_VALUE_FUNC(ilmErrorTypes, ilmControl_registerShutdownNotification, shutdownNotificationFunc, void*);
+}
+
+class IlmCommonTest : public ::testing::Test 
+{
+public:
+    void SetUp()
+    {
+        CLIENT_API_FAKE_LIST(RESET_FAKE);
+        RESET_FAKE(ilmControl_init);
+        RESET_FAKE(ilmControl_destroy);
+        RESET_FAKE(ilmControl_registerShutdownNotification);
+    }
+
+    void TearDown()
+    {
+        if(ilm_isInitialized())
+        {
+            ilm_destroy();
+        }
+    }
+
+    void mock_ilmInitSuccess()
+    {
+        mpp_wlDisplays[0] = (struct wl_display*)&m_wlDisplayFakePointer;
+        SET_RETURN_SEQ(wl_display_connect, mpp_wlDisplays, MAX_NUMBER);
+
+        mp_ilmErrorType[0] = ILM_SUCCESS;
+        SET_RETURN_SEQ(ilmControl_init, mp_ilmErrorType, MAX_NUMBER);
+    }
+
+    static constexpr uint8_t MAX_NUMBER = 1;
+    uint8_t m_wlDisplayFakePointer = 0;
+    struct wl_display* mpp_wlDisplays [MAX_NUMBER] = {nullptr};
+    ilmErrorTypes mp_ilmErrorType[MAX_NUMBER] = {ILM_FAILED};
+};
+
+/** ================================================================================================
+ * @test_id             ilm_init_cannotGetDisplay
+ * @brief               A failure of ilm_init() when wl_display_connect() returns a null object
```

Comment:
I would rephrase this to: Test case of ilm_init() where wl_display_connect fails, returns null object